src/Microsoft.Extensions.Http/DefaultHttpClientFactory.cs

```diff
@@ -1,17 +1,77 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using System.Net.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using System.Collections.Concurrent;
+using System.Threading;
 
 namespace Microsoft.Extensions.Http
 {
-    // This is just a starter implementation. It's obviously wrong on purpose.
-    // The plan is to get a basic sample in first, and improve this along with the sample.
     internal class DefaultHttpClientFactory : HttpClientFactory
```

Comment:
This class:
- Creates the actual client 
- Does caching
- Applies the options boilerplate

The only real reason to replace/extend this is if you want a different policy for how to create the client and do caching.